File: Binary Search/binarysearch.js
function binarySearch(arr, val) {
    var l = 0
    var h = arr.length - 1

    while (h > l) {
        var m = Math.floor((h + l) / 2)
        if (arr[m] == val) {
            return m
        }
        else if (arr[m] > val) {
            h = m - 1
        }
        else {
            l = m + 1
        }
    }
    return -1
}

File: Binary Search/binarysearch.py
def binary_search(arr, val):
    high = len(arr) - 1
    low = 0

    while high > low:
        mid = (high + low) // 2
        if arr[mid] == val:
            return mid
        elif arr[mid] > val:
            high = mid - 1
        else:
            low = mid + 1

    return -1
